### DISCUSSION TITLE
M5Stack ADS110 I2C A->D module setting gain?

### DISCUSSION THREAD

**Discussion by ashimotok0** [micropython/micropython#10466] [Hardware & Peripherals]
Using M5Stack ADS110  I2C  A->D modules.

Any ideas how you set the gain value please . Preferably using UIFlow or failing that Micropython

Module  & chip Datasheets describes 4 gain settings.

https://shop.m5stack.com/products/adc-unit

**Comment by 2dof** [comment]
see: 
https://github.com/ControlEverythingCommunity/ADS1100/blob/master/Python/ADS1100.py

example is for python smbus lib for I2C but it has good example for set/select configuration  of adc, based on datasheet You will be able to configure adc 

in micropython something like: 

from machine import I2C
...
i2c.writeto(ADC_ADRESS , CONFIG)

where:
ADC_ADRES  - adress of adc 
CONFIG:   1 byte data     (see TABLE V. Configuration Register   in data sheet : https://www.ti.com/lit/ds/symlink/ads1100.pdf 

 for example gain = 1 , bits [ PGA0, PGA0 ] = 0 , 0   then  CONFIG = b'0'   ( gain = 2  ->CONFIG = b'1' , gain = 4  -> CONFIG = b'2' 

 From UIflow  repo (https://github.com/m5stack/UIFlow-Code)  based on i2c_bus.py  lib  You Can Use easyI2C.write_u8() method
( see some examples https://community.m5stack.com/topic/1147/external-libraries-and-i2c-via-uiflow/5)
